### ISSUE TITLE
UniversalSentenceEncoder Error.

### ISSUE THREAD

**Issue by franklydog** [JohnSnowLabs/spark-nlp#2813]
## Description

The pipeline is DocumentAssembler->UniversalSentenceEncoder->UniversalSentenceEncoder， I use it classify text. When I moniter spark status。I found some task have errors.
I build a cluster with 3 servers(intel CPU 20 cores(40 threads), 128G memory size). When I run it with "spark-submit --packages com.johnsnowlabs.nlp:spark-nlp_2.11:2.7.5  --master yarn --deploy-mode client....."，It‘s throw a error. But when I  run it with "spark-submit --packages com.johnsnowlabs.nlp:spark-nlp_2.11:2.7.5  --master local[*]....." it seems OK. Is the USE module support yarn mode? 

## Expected Behavior

No task error.
## Current Behavior

whe run the spark programm on yarn, the error is following:
org.apache.spark.SparkException: Failed to execute user defined function($anonfun$dfAnnotate$1: (array<array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>>) => array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.serializefromobject_doConsume_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$11$$anon$1.hasNext(WholeStageCodegenExec.scala:624)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at org.apache.spark.rdd.RDD$$anonfun$foreach$1$$anonfun$apply$27.apply(RDD.scala:927)
	at org.apache.spark.rdd.RDD$$anonfun$foreach$1$$anonfun$apply$27.apply(RDD.scala:927)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2121)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2121)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$11.apply(Executor.scala:407)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1408)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:413)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Multiple OpKernel registrations match NodeDef '{{node module/text_preprocessor/SentencepieceEncodeSparse}}': 'op: "SentencepieceEncodeSparse" device_type: "CPU" constraint { name: "out_type" allowed_values { list { type: DT_INT32 } } }' and 'op: "SentencepieceEncodeSparse" device_type: "CPU" constraint { name: "out_type" allowed_values { list { type: DT_INT32 } } }'
	 [[module/text_preprocessor/SentencepieceEncodeSparse]]
	at org.tensorflow.Session.run(Native Method)
	at org.tensorflow.Session.access$100(Session.java:48)
	at org.tensorflow.Session$Runner.runHelper(Session.java:326)
	at org.tensorflow.Session$Runner.run(Session.java:276)
	at com.johnsnowlabs.ml.tensorflow.TensorflowWrapper.getTFHubSession(TensorflowWrapper.scala:109)
	at com.johnsnowlabs.ml.tensorflow.TensorflowUSE.calculateEmbeddings(TensorflowUSE.scala:40)
	at com.johnsnowlabs.nlp.embeddings.UniversalSentenceEncoder.annotate(UniversalSentenceEncoder.scala:142)
	at com.johnsnowlabs.nlp.AnnotatorModel$$anonfun$dfAnnotate$1.apply(AnnotatorModel.scala:35)
	at com.johnsnowlabs.nlp.AnnotatorModel$$anonfun$dfAnnotate$1.apply(AnnotatorModel.scala:34)
	... 20 more
## Possible Solution

No
## Steps to Reproduce

1.download https://nlp.johnsnowlabs.com/2020/12/08/tfhub_use_multi_lg_xx.html.
2.load the files. val use = UniversalSentenceEncoder.load("USE_Pretrained/tfhub_use_multi_lg_xx_2.7.0_2.4_1607439900967/")
.setInputCols("document")
.setOutputCol("sentence_embeddings")
3.pipeline = new Pipeline().
setStages(Array(
documentAssembler,
use,
classifierDL
))
4.pipeline.fit(data).
5. submit to spark on yarn . "spark-submit --packages com.johnsnowlabs.nlp:spark-nlp_2.11:2.7.5  --master yarn --deploy-mode client  --driver-memory 16g --executor-memory 10G --num-executors 20  --executor-cores 4  --conf spark.default.parallelism=500 --class ..."

## Your Environment

* Spark NLP version `sparknlp.version()`:spark-nlp_2.11:2.7.5
* Apache NLP version `spark.version`:spark 2.4
* Java version `java -version`:1.8
* Setup and installation (Pypi, Conda, Maven, etc.):Maven
* Operating System and version:Centos 7.8
* Link to your project (if any):

**Comment by maziyarpanahi** [comment]
same as https://github.com/JohnSnowLabs/spark-nlp/issues/2812 - the OS doesn't support multi-lingual USE models

**Comment by franklydog** [comment]
Thanks.But which os is supported by multi-lingual USE models?

**Comment by franklydog** [comment]
But when  spark submit with local mode it seems ok. when  Sparksubmit with yarn mode it 's throw error.

**Comment by maziyarpanahi** [comment]
This is a piece of new information that was not in the issue template. if the template is not completed it will be very hard to understand the issue.

Please:
- complete the template, when the issue happens and when it doesn't
- clear way for us to reproduce it, this means explaining what is your cluster, where it is, how you run it, the code you run on the cluster
- and please avoid creating new issues with the same topic, let's stay on this issue so we can follow or this issue will never resolve

The more info the better!

**Comment by cccs-is** [comment]
@franklydog Are you executors configured with multiple cores?
We ran into the same issue on a spark/k8s deployment and I found that running with single core executors avoided the error.

**Comment by maziyarpanahi** [comment]
That is a very bad advice! The whole point of having multiple executors on multiple machines is to scale in parallel. Having 1 core is a really bad scale.

If you are experiencing a problem, please open a new issue, complete the template step by step for us to reproduce the error and we will advice or fix any problems. (I personally use UncersalSentenceEncoder on Databricks and Cloudera over 20 machines with executor has 8-16 cores)

**Comment by cccs-is** [comment]
@maziyarpanahi I guess you misread my suggestion, I do have multiple executors, it's just that each is running single threaded (1 core each) and that does scale. In our tests we are using 16 to 32 executors. 
As you know, there are many ways of scaling, multiple machines (in our case worker pods on kubernetes), multiple executors (running on worker nodes) and multiple threads per executor, which might have a global state that could be shared.

**Comment by maziyarpanahi** [comment]
sounds fair, however, you shouldn't change your cluster setup to go around an issue with a third-party library. 

Please do report if there is an issue with Spark NLP and we either help or fix it.
